tests/acceptance/features/bootstrap/NotificationContext.php

```diff
@@ -134,4 +153,96 @@ public function theJsonDataFromLastResponseShouldMatch(
 			$this->featureContext->getJSONSchema($schemaString)
 		);
 	}
+
+	/**
+	 * @Then /^the email address of user "([^"]*)" should have received an email for space "([^"]*)" from user "([^"]*)" with the body containing$/
```

Comment:
wherever possible use the simple expression
```suggestion
	 * @Then the email address of user :user should have received an email for space :spaceName from user :sender with the body containing$/
```